index.bs

```diff
@@ -3253,14 +3253,35 @@ extended attribute’s [=takes a named argument list|named argument list=].
 </div>
 
 Two types are <dfn id="dfn-distinguishable" export>distinguishable</dfn> if
-at most one of the two [=includes a nullable type=], is a [=dictionary type=], or a [=record type=],
-and at least one of the following three conditions is true:
+the following algorithm returns <i>true</i>.
 
-1.  The two types (taking their [=inner types=]
+<ol class="algorithm">
+1.  If both types are either a [=union type=] or nullable union type,
+    return <i>true</i> if each member type of the one
+    is distinguishable with each member type of the other,
+    or <i>false</i> otherwise.
+1.  If one type is a [=union type=] or nullable union type,
+    return <i>true</i> if each
+    [=member type=] of the union type is distinguishable
+    with the non-union type,
+    or <i>false</i> otherwise.
+1.  If both types are any of:
+    *  a [=dictionary type=],
+    *  a [=record type=], or
+    *  a [=nullable type=],
```

Comment:
Because the previous two steps unwrap unions, I believe I don't need to run [includes a nullable type](https://heycam.github.io/webidl/#dfn-includes-a-nullable-type). Let me know if I'm wrong.